libraries/DataFlash/DataFlash_File.cpp

```diff
@@ -429,6 +430,39 @@ char *DataFlash_File::_log_file_name(const uint16_t log_num) const
 }
 
 /*
+  construct a log file name given a log number.
+  The number in the log filename will be zero-padded.
+  Note: Caller must free.
+ */
+char *DataFlash_File::_log_file_name_long(const uint16_t log_num) const
+{
+    char *buf = nullptr;
+    if (asprintf(&buf, "%s/%08u.BIN", _log_directory, (unsigned)log_num) == 0) {
```

Comment:
@muramura Wow, well spotted.

In my defense, I was just being *consistently* wrong with what was already in the file :-)

I've pushed up a patch beneath this one to fix the pre-existing problems, and corrected this one to not include the problem.

Thanks!